include/alpaka/block/shared/st/BlockSharedMemStOmp5.hpp

```diff
@@ -44,12 +44,20 @@ namespace alpaka
             //-----------------------------------------------------------------------------
             static auto declareVar(BlockSharedMemStOmp5 const& smem) -> T&
             {
-#    pragma omp barrier
-#    pragma omp single
+                auto* data = blockSharedMemSt.template getVar<T>(TuniqueId);
+
+                if(!data)
                 {
-                    smem.alloc<T>();
+#    pragma omp barrier
+#    pragma onmp single
+                    {
+                        blockSharedMemSt.template alloc<T>(TuniqueId);
+                    }
+                    // lookup for the data chunk allocated by the master thread
+                    data = blockSharedMemSt.template getVar<T>(TuniqueId);
                 }
-                return smem.getLatestVar<T>();
+                ALPAKA_ASSERT(data != nullptr);
```

Comment:
This assert should be guarded by a
```c++
#ifdef ALPAKA_DEBUG_OFFLOAD_ASSUME_HOST
```
otherwise it will break offloading in debug builds.

Maybe we need an `ALPAKA_ASSERT_OFFLOAD` or `ALPAKA_ASSET_DEVICE` which does this.

Maybe this guard should be